components/nexus-mime/src/main/java/org/sonatype/nexus/mime/MimeSupport.java

```diff
@@ -33,10 +35,28 @@
    * but not so precise as detection, where content is needed. This method should be used whenever a MIME type of a
    * file item that is contained <b>within</b> given context that has MimeRulesSource is about to be guessed.
    *
+   * @param mimeRulesSource optionally, the mimeRulesSource to apply/mixin into results.
    * @param path to guess for.
    * @return the most applicable MIME type as String.
    */
-  String guessMimeTypeFromPath(MimeRulesSource mimeRulesSource, String path);
+  @Nonnull
+  String guessMimeTypeFromPath(@Nullable MimeRulesSource mimeRulesSource, String path);
```

Comment:
Minor: you could use varargs here `String guessMimeTypeFromPath(String path, MimeRulesSource... mimeRulesSource)` to avoid having to repeat methods for when you have a rules source vs when you don't.

Also it looks like the primary client of this API only ever calls `guessMimeTypesListFromPath` so do we really need to maintain all these extra convenience methods which aren't currently used? Should be easy to add them back if they're needed in the future.